File: BOR2/apps/api/internal/domain/fuel.go
package domain

import "time"

type FuelEventType string

const (
	FuelEventTypeTrip FuelEventType = "trip"
	FuelEventTypeIdle FuelEventType = "idle"
)

type SamsaraEvent struct {
	ID         string        `json:"id"`
	EventKey   string        `json:"eventKey"`
	EventDate  time.Time     `json:"eventDate"`
	DriverName string        `json:"driverName"`
	Location   string        `json:"location"`
	Distance   float64       `json:"distance"`
	Units      string        `json:"units"`
	Type       FuelEventType `json:"type"`
	CreatedAt  time.Time     `json:"createdAt"`
}

type WexTransaction struct {
	ID              string    `json:"id"`
	TransactionKey  string    `json:"transactionKey"`
	TransactionDate time.Time `json:"transactionDate"`
	DriverName      string    `json:"driverName"`
	Units           float64   `json:"units"`
	TotalFuelCost   float64   `json:"totalFuelCost"`
	MerchantCity    string    `json:"merchantCity"`
	CreatedAt       time.Time `json:"createdAt"`
}

type FuelFilters struct {
	DriverName string
	StartDate  *time.Time
	EndDate    *time.Time
}

type FuelSummary struct {
	TotalDistance float64            `json:"totalDistance"`
	TotalFuelCost float64            `json:"totalFuelCost"`
	TotalIdleHours float64           `json:"totalIdleHours"`
	ByDriver      map[string]float64 `json:"byDriver"`
}

File: BOR2/apps/web/src/types/shared/fuel.ts
export type FuelEventType = "trip" | "idle"

export interface SamsaraEvent {
  id: string
  eventKey: string
  eventDate: string
  driverName: string
  location: string
  distance: number
  units: string
  type: FuelEventType
  createdAt: string
}

export interface WexTransaction {
  id: string
  transactionKey: string
  transactionDate: string
  driverName: string
  units: number
  totalFuelCost: number
  merchantCity: string
  createdAt: string
}

export interface FuelSummary {
  totalDistance: number
  totalFuelCost: number
  totalIdleHours: number
  byDriver: Record<string, { distance: number; cost: number }>
}

export interface FuelFilters {
  driverName?: string
  startDate?: string
  endDate?: string
}
